docs/season-plan.md

```diff
@@ -0,0 +1,149 @@
+ 1. Starting on the right side of the 2-ball tarmac.
+ 2. Starting on the left side of the 2-ball tarmac.
+ 3. Starting on the right side of the 1-ball tarmac.
+ 4. Starting on the left side of the 1-ball tarmac.
+- Document how to select which auto routine will run.
+- Create a way to start auto mode.
+- Place the robot on one of four starting positions and successfully run the corresponding auto routine.
+- Put the robot in teleop mode 15 seconds after game start.
+
+## Milestone 3 - "Move to All the Balls Auto"
+
+The game starts with robots in the tarmac and balls stationary on the ground at set positions. "Zero-ball" and "One-ball" auto don't require picking up any additional balls, but all other autos do. The first step in picking up additional balls is moving to the correct location.
+
+- Create an auto routine that moves the robot autonomously from the right or left side of the 2-ball tarmac in a set path that would allow it to pick up three balls (the wall ball, the field ball, and the terminal ball) within 15 seconds.
+- Document how to select which starting point.
+- Create a way to start auto mode.
+- Place the robot on one of two starting positions and successfully run the corresponding auto routine.
+- Put the robot in teleop mode after 15 seconds.
+
+## Milestone 4 - "Light Up One-Ball Auto"
+
+Until we have an intake and a shooting mechanism we can't run commands to control those subsystems on the robot. But we can issue commands to light up lights on the robot as a placeholder until those mechanisms are ready. "One-ball" auto requires issuing the following sequence of commands: shoot a ball, taxi out of the tarmac to the position we want to be in to start teleop mode, enter teleop mode after 15 seconds.
+
+- Create an auto routine that lights up lights on the robot to indicate when we would shoot a ball, then move the robot autonomously out of the tarmac from any of the four possible starting positions.
+- Document how to select which starting point.
+- Create a way to start auto mode.
+- Place the robot on one of four starting positions and successfully run the corresponding auto routine.
+- Put the robot in teleop mode after 15 seconds.
+
+# Autonomous Strategies
+
+See "Autos" doc.
+
+# Notes
+
+Point People (Students):
+Control Systems (Ivan)
+Auto (Jack)
+Vision (Gabe)
+Code Reviewers (Mentors):
+Dennis
+John Tower (Gabe's dad)
+Vision Plan:
+Limelight with targeting
+Limelight with position tracking: Robust Localization will need to utilize filtering.
+Jetson SLAM Localization
+Jetson SLAM as Object Detection
+Autonomous Plan:
+Pathweaver
+WPILib Trajectory
+Control Algorithms:
+Bang bang (start motor, stop motor)
+Heuristic (ramp up / ramp down, jello, etc.)
+PID control (proportional, integral, derivative)
+State space (physics model)
```

Comment:
LQR is an ideal controller/control scheme. State-space is just a representation of our physical system that takes an input and produces an output. You can create a feedback system with a reference, where K, in our case, is the output of LQR.
![image](https://user-images.githubusercontent.com/43253224/150059915-1524b60f-a524-47cf-97e5-fc4f0f30ab94.png)